src/component/structure-component.ts

```diff
@@ -201,6 +208,7 @@ class StructureComponent extends Component {
 
       this.rebuildRepresentations()
       this.rebuildTrajectories()
+      this.updateMatrix()
```

Comment:
This feels odd coupling matrix to selection - should the call instead be to `setPivot` explicitly if centerMode === 'selection' (which then triggers `updateMatrix` from the `setPivot` method itself)?  (I might just have misunderstood what's going on).

Reply:
When the selection changes, if the `centerMode` is "selection", the matrix has to be recalculated because the _center_ has changed, not the pivot point. So I don't think it would be clearer to call `setPivot(this.pivot.x, this.pivot.y, this.pivot.z)` rather than `this.updateMatrix()`. How about a comment there explaining why the matrix needs to be updated? And I will do it only when `centerMode === "selection"`.